File: rx-codelab/src/test/java/com/jraska/rx/codelab/SetupTest.java
package com.jraska.rx.codelab;

import com.jraska.rx.codelab.http.HttpModule;
import com.jraska.rx.codelab.http.RequestInfo;

import org.junit.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

public class SetupTest {
  @Test
  public void everythingSetUp() throws IOException {
    RequestInfo requestInfo = HttpModule.httpBinApi()
      .getRequest()
      .blockingFirst();

    assertThat(requestInfo.url).isNotNull();
    assertThat(requestInfo.origin).isNotNull();
    assertThat(requestInfo.headers).isNotEmpty();
  }
}

File: rx-codelab/src/test/java/com/jraska/rx/codelab/SetupTest.kt
package com.jraska.rx.codelab

import com.jraska.rx.codelab.http.HttpModule
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import java.io.IOException

class SetupTest {
  @Test
  @Throws(IOException::class)
  fun everythingSetUp() {
    val requestInfo = HttpModule.httpBinApi()
      .getRequest()
      .blockingFirst()

    assertThat(requestInfo.url).isNotNull()
    assertThat(requestInfo.origin).isNotNull()
    assertThat(requestInfo.headers).isNotEmpty
  }
}
